### ISSUE TITLE
Make column privileges matchable/callable

### ISSUE THREAD

**Issue by MichaelTiemannOSC** [os-climate/osc-trino-acl-dsl#1]
Rather than forcing users to enumerate every possible column name in the schema, column checks could be checked against a pattern (such as a prefix or regex) or even checked via a (lambda) function.

In this way, a USER1 could create and query all columns that their pattern matches (which might be the prefix '' or the pattern r'*').

USER2 could create and query all columns prefixed with 'quant_' and 'user_', but not '_dev'

USER3 could create and query only columns prefixed with 'user_'

This could extend beyond pattern matching to an arbitrary evaluation function, with all the performance and security problems contained therein.

Thoughts, @erikerlandson ?

**Comment by erikerlandson** [comment]
just to confirm, what you are asking is something like this:
```
    "columns": [
      { "name": "dev.*", "allow": false }
    ]
```

so asking trino to hide all columns whose name starts with "dev" ?

**Comment by MichaelTiemannOSC** [comment]
Yes, though technically I asked for "dev_*", which "dev.*" would indeed match (because . matches _).


Michael

________________________________

**Comment by erikerlandson** [comment]
Have not yet received a response about whether hidden columns can be given as regex yet
